bundles/org.openhab.binding.omnilink/src/main/java/org/openhab/binding/omnilink/internal/discovery/OmnilinkDiscoveryService.java

```diff
@@ -0,0 +1,545 @@
+                    int areaFilter = bitFilterForArea(areaProperties);
+
+                    ObjectPropertyRequest<ButtonProperties> objectPropertyRequest = ObjectPropertyRequest
+                            .builder(handler, ObjectPropertyRequests.BUTTONS, 0, 1).selectNamed().areaFilter(areaFilter)
+                            .build();
+
+                    for (ButtonProperties buttonProperties : objectPropertyRequest) {
+                        String thingName = buttonProperties.getName();
+                        String thingID = Integer.toString(buttonProperties.getNumber());
+
+                        Map<String, Object> properties = new HashMap<>();
+                        properties.put(THING_PROPERTIES_NAME, thingName);
+                        properties.put(THING_PROPERTIES_AREA, areaProperties.getNumber());
+
+                        ThingUID thingUID = new ThingUID(THING_TYPE_BUTTON, bridgeUID, thingID);
+
+                        DiscoveryResult discoveryResult = DiscoveryResultBuilder.create(thingUID)
+                                .withProperties(properties).withProperty(THING_PROPERTIES_NUMBER, thingID)
+                                .withRepresentationProperty(THING_PROPERTIES_NUMBER).withBridge(bridgeUID)
+                                .withLabel(thingName).build();
+                        thingDiscovered(discoveryResult);
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Discovers OmniLink locks
+     */
+    private void discoverLocks() {
+        final OmnilinkBridgeHandler handler = bridgeHandler;
+        if (handler != null) {
+            final ThingUID bridgeUID = handler.getThing().getUID();
+
+            ObjectPropertyRequest<AccessControlReaderProperties> objectPropertyRequest = ObjectPropertyRequest
+                    .builder(handler, ObjectPropertyRequests.LOCK, 0, 1).selectNamed().build();
+
+            for (AccessControlReaderProperties lockProperties : objectPropertyRequest) {
+                String thingName = lockProperties.getName();
+                String thingID = Integer.toString(lockProperties.getNumber());
+
+                Map<String, Object> properties = new HashMap<>();
+                properties.put(THING_PROPERTIES_NAME, thingName);
+
+                ThingUID thingUID = new ThingUID(THING_TYPE_LOCK, bridgeUID, thingID);
+
+                DiscoveryResult discoveryResult = DiscoveryResultBuilder.create(thingUID).withProperties(properties)
+                        .withProperty(THING_PROPERTIES_NUMBER, thingID)
+                        .withRepresentationProperty(THING_PROPERTIES_NUMBER).withBridge(bridgeUID).withLabel(thingName)
+                        .build();
+                thingDiscovered(discoveryResult);
+            }
+        }
+    }
+
+    /**
+     * Discovers OmniLink audio zones
+     */
+    private void discoverAudioZones() {
+        final OmnilinkBridgeHandler handler = bridgeHandler;
```

Comment:
Some of them it might be possible to, but because there are different supported properties and the way the Omni processes the requests, it would be difficult to do. As such, I would prefer to leave it for now.